tests/suites/test_suite_psa_crypto_persistent_key.data

```diff
@@ -1,26 +1,26 @@
 
 Parse storage: truncated magic
-parse_storage_data_check:"505341004b4559":"":PSA_KEY_LIFETIME_PERSISTENT:PSA_KEY_TYPE_RSA_KEY_PAIR:PSA_KEY_USAGE_EXPORT:PSA_ALG_CATEGORY_ASYMMETRIC_ENCRYPTION:PSA_ALG_CATEGORY_SIGN:PSA_ERROR_STORAGE_FAILURE
+parse_storage_data_check:"505341004b4559":"":PSA_KEY_LIFETIME_PERSISTENT:PSA_KEY_TYPE_RSA_KEY_PAIR:0:PSA_KEY_USAGE_EXPORT:PSA_ALG_CATEGORY_ASYMMETRIC_ENCRYPTION:PSA_ALG_CATEGORY_SIGN:PSA_ERROR_STORAGE_FAILURE
 
```

Comment:
Please add negative test cases where the key data is inconsistent with the bit-size. Given the level of coverage here, one AES key with `bits != data_length * 8` and one RSA key with valid data and the wrong size would be good.

Reply:
What would you expect to happen in such a case? Currently, I kept the old import logic upon loading transparent keys, since the wording in the documentation was that the transparent keys are stored in the exported format, which might be different than the internal representation. This import logic also computes and sets the bits attribute value.

For transparent keys the tests you are suggesting would not be affected by the stored bits value, but tests using opaque keys would depend on it.

Reply:
Since the loading code reuses the import logic, I'd expect it to hit the [check where if there's an explicit key size, it's compared against the actual key size](https://github.com/ARMmbed/mbedtls/blob/mbedtls-2.24.0/library/psa_crypto.c#L2061).

Reply:
I am still not completely sure what you mean here. The test the comment refers to (`parse_storage_data_check`) exercises `psa_parse_key_data_from_storage` directly, which only unpacks the storage format. It doesn't really perform any validation of the content, since it doesn't necessarily know how it should validate it.

The existing import logic is only used in the higher level import/export tests. When re-loading a key, `psa_import_key_into_slot` is triggered, but not `psa_validate_optional_attributes`. Attribute validation is only performed on import.

Reply:
Oh, I see. I hadn't realized that the test code only had limited ability to filter bad inputs. This is only a minor test gap since the storage is only supposed to contain what (a previous version of) Mbed TLS put there. So I withdraw this request.